### ISSUE TITLE
Markdig version

### ISSUE THREAD

**Issue by PassivePicasso** [gwaredd/UnityMarkdownViewer#15]
I've been working on a UIToolkit implementation of Markdig, and I noticed that the Markdig distribution of the dll doesn't work in Unity. 

Did you use a custom version of markdig to make it work in unity?

**Comment by gwaredd** [comment]
That's a good question - maybe, I recall there was some issues that needed resolving but I did that a while ago and I can't recall the details. What are the errors you are seeing?

Unity also seem to have [a custom version](https://github.com/Unity-Technologies/markdig)

**Comment by PassivePicasso** [comment]
Markdig relies on system.memory which isn't available unfortunately.

Thats the only issue I saw, supposedly thats been that way since version 18

**Comment by PassivePicasso** [comment]
Seems like perhaps the UT version points at that specific version.  I didn't verify but its pretty far out of date relative to the original repo

**Comment by gwaredd** [comment]
Looks like there are a few flavours of markdig for the various .net variants - e.g. net4.5, core2.1, standard2.1 etc.

If you clone the [repro](https://github.com/xoofx/markdig) and build it manually you can pick the one that works for you. The net4.5 includes the System.Memory.dll (the other versions don't appear to use it) which you can import into unity as well. Note you may need to flag any assemblies that use it as allow unsafe code.

**Comment by PassivePicasso** [comment]
I had downloaded the repository and actually tried all the variants, but Unity always complained about System.Memory.
In Hindsight, my tests were not robust, I was merely executing the build and then copying DLLs.
I never actually verified that the outputs were what I was expecting to get with switching the build versions.
I didn't flag my library for allowing unsafe, so its possible that could be related to why I was unsuccessful.

Last night I ended up switching to Markdig.Signed Version 18.3 from Nuget which appears to be the latest immediately functional version available on Nuget.  This did work well, but obviously it would be ideal to be on the latest.

Just in case your curious, this is my markdown implementation for UIToolkit here, its currently embedded in another project, but I plan on extracting it out into an independent project.  
It's implemented as with the Markdig Renderers setup like yours except it renders VisualElement objects rather than executing IMGUI code.
The UIToolkit side isn't a perfect implementation due the lack of a real rich text control for VisualElements, so every word is rendered as a Label which is far from ideal and causes performance problems for large documents, but works surprisingly well considering that poor implementation detail, its setup to be toggleable at the moment so you have a choice between layout quality and performance. I plan on circling back to it and trying to implement a lower level solution that deals with these problems.
https://github.com/PassivePicasso/ThunderKit/tree/package-manager/Editor/Markdown
